slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -81,6 +81,31 @@ void MapBuilder::SaveMapToFile(bool include_unfinished_submaps,
     }
 }
 
+std::string MapBuilder::SaveMapToStream(const std::string path_to_dir) {
+    std::string filename = path_to_dir + "/" + "temp_internal_state.pbstream";
+    SaveMapToFile(false, filename);
+
+    std::ifstream tempFile(filename);
+    if (tempFile.bad()) {
+        LOG(ERROR) << "Failed to open " << filename << "as ifstream obejct.";
```

Comment:
Do we really want to return a blank buffer if we fail to read the file? Shouldn't this be a GRPC error?

Reply:
No you are correct, we should return an error from `GetInternalState` if a blank buf is produced by `SaveMapToStream`